projects/controllable_dialogue/README.md

```diff
@@ -24,11 +26,254 @@ improvements in human quality judgments.
+      --dict-file data/models/controllable_dialogue/dict_twit30k_train_split \
+      -mf /path/to/your/modelfile \
+      --control-vars avg_nidf,question
+
+To take an existing non-CT model (e.g. the baseline) and the finetune it as a CT model do this:
+
+    python projects/controllable_dialogue/train_controllable_seq2seq.py \
+    --fromfile-datapath data/ConvAI2_controllable/train.txt \
+    --fromfile-datapath2 data/ConvAI2_controllable/valid.txt \
+    --dict-file data/models/controllable_dialogue/dict_twit30k_train_split \
+    -mf /path/to/your/modelfile \
+    --init-model data/models/controllable_dialogue/convai2_finetuned_baseline \
+    --add-control True \
+    -cv avg_nidf
+
+This command will take the parameters saved in `--init-model`, load them in the new model (which has randomly initialized weights for the new CT parameters), and then save that model to the given modelfile (`-mf`). It should be quick. Once that's done, run this command:
+
+    python projects/controllable_dialogue/train_controllable_seq2seq.py \
+    --fromfile-datapath data/ConvAI2_controllable/train.txt \
+    --fromfile-datapath2 data/ConvAI2_controllable/valid.txt \
+    --dict-file data/models/controllable_dialogue/dict_twit30k_train_split \
+    -mf /path/to/your/modelfile \
+    --add-control False \
+    -cv avg_nidf
+
+You should see your new CT model training. Note: this is how the models in the paper were trained.
+
+## Look at output and automatic metrics
+
+First, download and extract [this file](https://drive.google.com/open?id=16kXJXJ8WiSrUpAsShYcuzJhToVBrnf30) to obtain a directory called `wordstat_files`. The directory contains json files with the generated output and the automatic metrics computed for the pretrained models.
+
+Run the following:
+
+    cd projects/controllable_dialogue
+    jupyter notebook
+
+and then open up `inspect_wordstats.ipynb`. Where it says `models_dir`, enter the path to your `wordstat_files` directory. You will be able to recreate the table of automatic metrics from the paper (Table 6), and you can explore the output of the models.
+
+TODO: <screenshot>
+
+## Measure automatic metrics
+
+If you want to generate json files like those in the previous section, do this:
+
+    cd ParlAI
+    python projects/controllable_dialogue/eval_wordstat.py \
+    --fromfile_datapath data/ConvAI2_controllable/valid.txt \
+    -mf data/models/controllable_dialogue/control_questionb11e10 \
+    -wd extrep_2gram:-3.5,extrep_nonstopword:-1e20,intrep_nonstopword:-1e20 \
+    --set-controls question:7
+
+This will create a json file containing the output and other automatic metrics for the question-controlled CT model with z=7. The script `eval_wordstat.py` always places the json file in the same place as the model file. The script can take a while to complete.
+
+Note: Due to changes in ParlAI, there might be some small differences between the json file created via this method, and the json files downloadable in the previous section.
```

Comment:
Is this presently true?

Reply:
Which part? The first paragraph is true yes. The part about small differences - I haven't exhaustively checked because regenerating all the json files takes a long time, but I've generated some, and the important things (e.g. the Table 6 metrics) are the same. 

There are small differences in the arora.pkl and word2count.pkl files when I regenerate them (I haven't yet done a deep dive to find out why), and this can cause further differences when running the model, hence why I'm noting that there could be a difference in the json files here.